### ISSUE TITLE
Action Breaks with Nested Branch Names

### ISSUE THREAD

**Issue by chris-hahn-kr** [Sauloxd/stories#8]
Hi, there!

We are using this action to deploy our review apps for Pull Requests and it is working quite nicely for the most part. The action runs and creates the new Github page successfully.

The problem comes right after the folder is successfully created in our `gh-pages` branch. It's trying to fetch the origin of the original PR branch and fails.

At first glance, it looks like it is only grabbing the branch name after the `/`.

For example... we have a branch called `feat/migrate-to-github-actions`. Creating a PR against our main branch, everything works perfectly fine. All of our tests run, the review app kicks off, and then creates our new folder in our `gh-pages` directory.

Then the action throws an error, because it's trying to fetch origin on `migrate-to-github-actions` and not `feat/migrate-to-github-actions` where our actual branch is. A real example error encounter and output can be seen below:

```
/usr/bin/git push origin gh-pages
To https://github.com/orgNameRedacted/repoNameRedacted
   86a2253..a75c6a5  gh-pages -> gh-pages
/usr/bin/git fetch origin migrate-to-github-actions
fatal: couldn't find remote ref migrate-to-github-actions
Error: The process '/usr/bin/git' failed with exit code 128
(node:14081) UnhandledPromiseRejectionWarning: Error: The process '/usr/bin/git' failed with exit code 128
    at ExecState._setResult (/home/vsts-agent/actions-runner/_work/_actions/sauloxd/review-apps/v1.3.3/packages/action/dist/index.js:1026:25)
    at ExecState.CheckComplete (/home/vsts-agent/actions-runner/_work/_actions/sauloxd/review-apps/v1.3.3/packages/action/dist/index.js:1009:18)
    at ChildProcess.<anonymous> (/home/vsts-agent/actions-runner/_work/_actions/sauloxd/review-apps/v1.3.3/packages/action/dist/index.js:903:27)
    at ChildProcess.emit (events.js:210:5)
    at maybeClose (internal/child_process.js:1021:16)
    at Socket.<anonymous> (internal/child_process.js:430:11)
    at Socket.emit (events.js:210:5)
    at Pipe.<anonymous> (net.js:659:12)
(node:14081) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:14081) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

(I redacted org and repo name above since the is for a private setup)

**Comment by mDuo13** [comment]
I have observed [a similar problem in a repo I work on](https://github.com/ripple/xrpl-dev-portal/runs/2774891954?check_suite_focus=true). The error message is different, but so far I've only seen it occur on branches with `/` in the name:

```
1 Run sauloxd/review-apps@v1.3.3
11 /usr/bin/git config --global user.name review-app-action-user
12 /usr/bin/git config --global user.email review-app-action-email
13 (node:2397) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'substr' of undefined
14     at createReviewApp (/home/runner/work/_actions/sauloxd/review-apps/v1.3.3/packages/action/dist/index.js:10167:60)
15     at processTicksAndRejections (internal/process/task_queues.js:93:5)
16     at async /home/runner/work/_actions/sauloxd/review-apps/v1.3.3/packages/action/dist/index.js:10206:5
17 (node:2397) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of  an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
18 (node:2397) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
19 /usr/bin/git config pull.rebase true
20 Error: Cannot read property 'substr' of undefined
```

**Comment by Sauloxd** [comment]
Fixed in #14  !
I made a huge refactor in this PR together with this fix, the issue was that I was getting the branch name information from a wrong source! Hope it works for you guys now
